person.rb

```diff
@@ -1,26 +1,85 @@
-require_relative 'nameable'
+class Nameable
+  attr_accessor :name
+
+  def initialize(name)
+    @name = name
+  end
+
+  def correct_name
+    raise NotImplementedError, 'Kindly Implement correct_name method to return the correct name'
+  end
+end
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
 
-  def initialize(age, name: 'Unknown', parent_permission: true)
-    super()
-    @id = Random.rand(1..1000)
+  def initialize(age, name)
+    super(name)
+
     @name = name
+
     @age = age
-    @parent_permission = parent_permission
   end
 
   def can_use_services?
-    of_age? || @parent_permission
+    if is_of_age? || @parent_permission
+
+      true
+
+    else
+
+      false
+
+    end
   end
 
   def correct_name
     @name
-ate
+  end
+
+  private
 
   def of_age?
     @age >= 18
   end
 end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
+
+    @nameable = nameable
+  end
```

Comment:
- [ ] While you made the changes requested by the previous reviewer, You removed the `correct_name` method in BaseDecorator, so kindly add it again